site/pages/api/pledge/index.ts

```diff
@@ -1,10 +1,41 @@
 import { findOrCreatePledge } from "components/pages/Pledge/lib/firebase";
 import { NextApiRequest, NextApiResponse } from "next/types";
+import { array, boolean, number, object, string } from "yup";
 
 export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse
 ) {
+  const pledgeSchema = object({
```

Comment:
Ah so this was missed in the overall comment.
Let's use the schema we've already defined for pledge form. I expect both shapes to be exactly the same so better not to duplicate this.

Only caveat I can think of is translations. We return a lingui translation id string so that probably needs to be handled on the FE.

Reply:
@0xAeterno I hear ya but the schema doesnt work in pages/api because of the `isOwner` check using window. We get the error "window is not defined" bc its on the server. Also not sure if this will delete the updatedAt with the noUnknown check or if thats added by firebase (havent checked bc of window issue)

Reply:
Gotcha @goatsOnDrugs.
What's the purpose of `isOwner`?, IMO the formSchema should be independent from dependancies like the `window` object.
`isOwner` feels like a validation that should belong on the server, but also there's probably another way to approach how we validate here.

Can you please create a ticket with some context for us to look at this later. I'm okay with coming back to this later

Reply:
isOwner checks that the user isnt adding the pledge owner address as a secondary wallet. I can create a ticket for this right meow.

Reply:
TY for the review!